arch/risc-v/src/rv64gc/riscv_assert.c

```diff
@@ -255,18 +258,48 @@ static void up_dumpstate(void)
   _alert("stack size: %016" PRIxPTR "\n", ustacksize);
 #endif
 
+#ifdef CONFIG_ARCH_KERNEL_STACK
+  /* Does this thread have a kernel stack allocated? */
+
+  if (rtcb->xcp.kstack)
+    {
+      kstackbase = (uintptr_t)rtcb->xcp.kstack;
+
+      _alert("Kernel stack:\n");
+      _alert("  base: %016\n" PRIxPTR "\n", kstackbase);
+      _alert("  size: %016\n" PRIxPTR "\n", CONFIG_ARCH_KERNEL_STACKSIZE);
+    }
+#endif
+
   /* Dump the user stack if the stack pointer lies within the allocated user
    * stack memory.
    */
 
   if (sp >= ustackbase && sp < ustackbase + ustacksize)
     {
-      _alert("ERROR: Stack pointer is not within allocated stack\n");
```

Comment:
This is still an error for kernel threads

Reply:
I don't think so, the "ustack" in this case should be the primary stack allocated for the task. Unless I misunderstand something ? Do the kernel tasks share one stack or something ?